webpack/assets/javascripts/react_app/components/Pagination/index.js

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useHistory } from 'react-router-dom';
+import URI from 'urijs';
+import classNames from 'classnames';
+import {
+  Pagination as PF4Pagination,
+  PaginationVariant,
+} from '@patternfly/react-core';
+import { translate as __ } from '../../common/I18n';
+import {
+  getURIpage,
+  getURIperPage,
+  changeQuery,
+} from '../../common/urlHelpers';
+import { useForemanSettings } from '../../Root/Context/ForemanContext';
+import './index.scss';
+
+const Pagination = ({
+  itemCount,
+  onSetPage,
+  onPerPageSelect,
+  className,
+  page: propsPage,
+  perPage: propsPerPage,
+  noSidePadding,
+  ...props
+}) => {
+  const { perPage: settingsPerPage } = useForemanSettings();
+  const history = useHistory();
+  const page = getURIpage() || propsPage;
+  const urlPerPage = getURIperPage();
+  const perPage = urlPerPage || propsPerPage || settingsPerPage;
+
+  const paginationTitles = {
+    items: __('items'),
+    page: __('page'),
+    itemsPerPage: __('Items per page'),
+    perPageSuffix: __('per page'),
+    toFirstPage: __('Go to first page'),
+    toPreviousPage: __('Go to previous page'),
+    toLastPage: __('Go to last page'),
+    toNextPage: __('Go to next page'),
+    optionsToggle: __('Items per page'),
+    currPage: __('Current page'),
+    paginationTitle: __('Pagination'),
+  };
+
+  const getPerPageOptions = () => {
+    const options = new Set([5, 10, 15, 25, 50]);
+    options.add(settingsPerPage);
+    if (urlPerPage) options.add(urlPerPage);
+    return [...options]
+      .sort((a, b) => a - b)
+      .map(value => ({ title: value.toString(), value }));
+  };
+
+  const _onSetPage = (e, nextPage) => {
+    if (onSetPage) return onSetPage(nextPage);
+    return updateSearch({ page: nextPage });
+  };
+
+  const _onPerPageSelect = (e, nextPerPage) => {
+    if (onPerPageSelect) return onPerPageSelect(nextPerPage);
+    return updateSearch({ per_page: nextPerPage, page: 1 });
+  };
+
+  const updateSearch = params => {
+    if (history) {
+      const uri = new URI();
+      uri.setSearch(params);
+      history.push({ search: uri.search() });
+    }
+    changeQuery(params);
+  };
+
+  const cx = classNames('tfm-pagination', className, {
+    'no-side-padding': noSidePadding,
```

Comment:
I think this should be done in the parent component that will use the pagination.
for example:
``` js
<Pagination
    classname='no-side-padding' 
...
```

Reply:
imho it makes it easier to use, since it's boolean value the consumer can pass it to the component as:
```js
<Pagination noSidePadding {...otherProps} />
```